crates/router/src/core/cards_info.rs

```diff
@@ -0,0 +1,31 @@
+use common_utils::fp_utils::when;
+use error_stack::{report, ResultExt};
+use router_env::{instrument, tracing};
+
+use crate::{
+    core::errors::{self, RouterResponse},
+    db::StorageInterface,
+    services::ApplicationResponse,
+    types::transformers::ForeignFrom,
+};
+
+#[instrument(skip_all)]
+pub async fn retrieve_card_info(
+    store: &dyn StorageInterface,
+    card_iin: String,
+) -> RouterResponse<api_models::cards_info::CardInfoResponse> {
+    when(card_iin.len() != 6, || {
+        Err(errors::ApiErrorResponse::InvalidCardIinLength)
+    })?;
```

Comment:
As far as I understand, length should be either 6 or 8 digits. Not sure if some cases need 9 chars as well.
So we can check if the length is in that range.

You'd have to update the error message as well, to convey the same.